fighters/dedede/src/acmd/specials.rs

```diff
@@ -346,42 +407,188 @@ unsafe extern "C" fn game_specialairlwstart(agent: &mut L2CAgentBase) {
+            let article = ArticleModule::get_article(agent.module_accessor, *FIGHTER_DEDEDE_GENERATE_ARTICLE_JETHAMMER);
+            let object_id = smash::app::lua_bind::Article::get_battle_object_id(article) as u32;
+            let article_boma = sv_battle_object::module_accessor(object_id);
+
+            MotionModule::change_motion(article_boma, Hash40::new("start"), 0.0, 1.0, false, 0.0, false, false);
+            MotionModule::set_frame(article_boma, 5.0, true);
+
+        }
+    }
+    frame(lua_state, 8.0);
+    FT_MOTION_RATE(agent, 1.0);
+    if is_excute(agent){
+        if ArticleModule::is_exist(agent.module_accessor, *FIGHTER_DEDEDE_GENERATE_ARTICLE_JETHAMMER){
+            let article = ArticleModule::get_article(agent.module_accessor, *FIGHTER_DEDEDE_GENERATE_ARTICLE_JETHAMMER);
+            let object_id = smash::app::lua_bind::Article::get_battle_object_id(article) as u32;
+            let article_boma = sv_battle_object::module_accessor(object_id);
+
+            MotionModule::change_motion(article_boma, Hash40::new("hold"), 0.0, 1.0, false, 0.0, false, false);
+
+        }
+    }
+    frame(lua_state, 9.0);
+    FT_MOTION_RATE(agent, 7.0 / (27.0-10.0));
+
+}
+
+unsafe extern "C" fn effect_specialairlwstart(agent: &mut L2CAgentBase) {
+    let lua_state = agent.lua_state_agent;
+    let boma = agent.boma();
+
+    frame(lua_state, 5.0);
+    if is_excute(agent){
+        EFFECT_FOLLOW(agent, Hash40::new("dedede_jethammer_hold"), Hash40::new("hammer3"), 0, 0, 0, 0, 0, 0, 1.0, true);
+    }
+    frame(lua_state, 8.0);
+    if is_excute(agent){
+        EFFECT_FOLLOW(agent, Hash40::new("dedede_jethammer_hold2"), Hash40::new("hammer3"), 0, 0, 0, 0, 0, 0, 1.0, true);
+    }
+}
+
+unsafe extern "C" fn sound_specialairlwstart(agent: &mut L2CAgentBase) {
+    let lua_state = agent.lua_state_agent;
+    let boma = agent.boma();
+
+    if is_excute(agent) {
+        PLAY_SE(agent, Hash40::new("se_dedede_special_l01"));
+    }
+    frame(lua_state, 5.0);
+    if is_excute(agent){
+        PLAY_STATUS(agent, Hash40::new("se_dedede_special_l02"));
+    }
+}
+
+unsafe extern "C" fn expression_specialairlwstart(agent: &mut L2CAgentBase) {
+    let lua_state = agent.lua_state_agent;
+    let boma = agent.boma();
+    
+    frame(lua_state, 2.0);
+    if is_excute(agent){
+        VisibilityModule::set_int64(boma, hash40("hammer") as i64, hash40("hammer_disp_off") as i64);
     }
 }
 
 unsafe extern "C" fn game_speciallw(agent: &mut L2CAgentBase) {
     let lua_state = agent.lua_state_agent;
     let boma = agent.boma();
-    let rush_speed = 0.6 + 0.01*WorkModule::get_float(boma, *FIGHTER_DEDEDE_STATUS_JET_HAMMER_WORK_FLOAT_HOLD_COUNT);
+    let charge_frames = VarModule::get_int(agent.battle_object, vars::dedede::instance::JET_CHARGE_FRAMES);
+    let charge_level = charge_frames as f32 / 30.0;
+
+    if is_excute(agent) {
+        FT_MOTION_RATE(agent, 45.0/(23.0));
+    }
```

Comment:
avoid using `FT_MOTION_RATE` or similar motion-rating functions *inside* an `if is_excute(agent)` conditional. This often causes problems for statuses that transition between air/ground motions when landing/falling, because the conditional fails in those circumstances

rewrite by just removing the conditional and keeping the motion rate